### ISSUE TITLE
java.lang.ClassCastException on throwables in FetcherBasedDeferredResolver

### ISSUE THREAD

**Issue by lgmyrek** [sangria-graphql/sangria#327]
result of Future.failed is Throwable
and i won't be matched to first clause (in some cases) of:
```scala
              case Some(e: Exception) ⇒ throw e
              case Some(v) ⇒ updateCache(id, v)
```
 but to second resulting in ClassCastException in following execution.

**Comment by OlegIlyenko** [comment]
I think you are right. `FetcherBasedDeferredResolver` should pattern match on `Throwable` instead of `Exception`. Would you mind making a PR?

**Comment by OlegIlyenko** [comment]
Fixed in #329
